ArduSub/AP_Arming_Sub.cpp

```diff
@@ -12,11 +12,45 @@ bool AP_Arming_Sub::rc_calibration_checks(bool display_failure)
     return rc_checks_copter_sub(display_failure, channels);
 }
 
+bool AP_Arming_Sub::has_disarm_function () {
+    bool has_shift_function = false;
+    // make sure the craft has a disarm button assigned before it is armed
+    // check all the standard btn functions
+    for (uint8_t i = 0; i < 16; i++) {
+        switch (sub.get_button(i)->function(false)) {
+            case JSButton::k_shift :
+                has_shift_function = true;
+                break;
+            case JSButton::k_arm_toggle :
+                return true;
```

Comment:
nitpick: could just fall through here (you do just that in the other loop).

Reply:
i thought about it, but i figured the point of using a switch was to make things visually explicit/easy to read.  If it just falls through it might as have just been the if statement IMO